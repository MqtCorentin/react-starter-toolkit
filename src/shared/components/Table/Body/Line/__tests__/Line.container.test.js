import React from 'react';
import { render } from '@testing-library/react';
import LineContainer from '../Line.container';

const defaultProps = {
  actions: null,
  children: null,
  idKey: 'uniqueid',
  cols: {},
};

const container = document.createElement('tbody');

const renderWithContainer = Component =>
  render(<Component />, {
    container: document.body.appendChild(container),
  });

describe('LineContainer', () => {
  it('Render <LineContainer/> without cols', () => {
    const { asFragment } = renderWithContainer(() => <LineContainer {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <LineContainer/> with cols and actions', () => {
    const ActionsLineContainer = ({ idKey, ...props }) => (
      <p id={idKey} {...props}>
        action body
      </p>
    );
    const { asFragment } = renderWithContainer(() => (
      <LineContainer
        {...defaultProps}
        cols={{
          name: {
            label: 'samuel',
          },
          phone: {
            label: '0606060606',
          },
        }}
        actions={ActionsLineContainer}
      />
    ));
    expect(asFragment()).toMatchSnapshot();
  });
});
