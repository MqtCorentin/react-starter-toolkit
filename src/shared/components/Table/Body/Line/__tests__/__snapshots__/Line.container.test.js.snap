// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineContainer Render <LineContainer/> with cols and actions 1`] = `
<DocumentFragment>
  <p
    id="uniqueid"
  >
    action body
  </p>
</DocumentFragment>
`;

exports[`LineContainer Render <LineContainer/> without cols 1`] = `<DocumentFragment />`;
